docs/performance-comparison/rosbag2-plugin-comparison.md

```diff
@@ -0,0 +1,152 @@
+- `mcap_default`: The default recording configuration of the MCAP storage plugin. Produces un-compressed, chunked MCAPs with a message index for efficient read performance.
+- `mcap_nochunking`: The highest-throughput configuration of the MCAP storage plugin. Does not write a message index to the MCAP, so bags recorded with this mode need to be reindexed later to be read efficiently.
+- `mcap_uncompressed_crc`: Like `mcap_default`, but calculates CRCs for each chunk so that readers can identify if a chunk contains corrupted data. This feature does not have an equivalent in the SQLite3 storage plugin.
+- `mcap_compressed_nocrc`: Like `mcap_default`, but compressed each chunk with `zstd` compression on default settings.
+
+## Software Versions
+
+Some installed package versions were omitted for clarity.
+
+| ROS 2 Package Name                | Version |
+| --------------------------------- | ------- |
+| mcap_vendor                       | 0.5.0   |
+| rcl                               | 5.4.0   |
+| rmw                               | 6.3.0   |
+| rmw_dds_common                    | 1.7.0   |
+| rmw_fastrtps_cpp                  | 6.3.0   |
+| rmw_fastrtps_shared_cpp           | 6.3.0   |
+| rmw_implementation                | 2.9.0   |
+| rmw_implementation_cmake          | 6.3.0   |
+| ros_core                          | 0.10.0  |
+| ros_environment                   | 3.2.0   |
+| ros_workspace                     | 1.0.2   |
+| rosbag2_compression               | 0.17.0  |
+| rosbag2_cpp                       | 0.17.0  |
+| rosbag2_storage                   | 0.17.0  |
+| rosbag2_storage_default_plugins   | 0.17.0  |
+| rosbag2_storage_mcap              | 0.5.0   |
+| rosbag2_storage_plugin_comparison | 0.1.0   |
+| rosbag2_storage_sqlite3           | 0.17.0  |
+| rosbag2_test_common               | 0.17.0  |
+| shared_queues_vendor              | 0.17.0  |
+| sqlite3_vendor                    | 0.17.0  |
+| zstd_vendor                       | 0.17.0  |
+
+## Message sizes tested
+
+- 1MiB
+- 10KiB
+- 100B
+- Mixed: Messages from all of the above sizes at the following ratios (by number of bytes, not messages)
+  - 1MiB: 70%
+  - 10KiB: 20%
+  - 100B: 10%
+
+When testing write throughput, 250MiB of messages are stored as quickly as possible.
+
+## Cache sizes tested
+
+The `rosbag2::SequentialWriter` collects messages in an internal cache before calling the storage plugin `write()` call with all messages at once. The cache size determines the size of each "batch" that is written to a storage plugin at once. This matters in particular for the SQLite storage plugin, because each batch constitutes a SQL transaction, and it's generally better for write performance to write many messages in one transaction.
+
+Cache sizes tested were:
+
+- 1KiB
+- 10MiB
```

Comment:
@james-rms You are mentioning that it was used cache size `1KiB` and `10Mib` for the `SequentialWriter`, but we are using number of messages not bytes for SequentialWriter's cache. 
It seems this is wrong statement in article.

Reply:
Will clarify - this benchmark tries to pack enough messages into each write() batch to meet a target size, not a target number of messages.

Reply:
Actually, I think i'm right here. The cache buffer decides when it's full based on a number of bytes, not a number of messages. Also, the cache size parameter [here](https://github.com/ros2/rosbag2/blob/rolling/rosbag2_cpp/src/rosbag2_cpp/cache/message_cache.cpp#L31) gets passed down to [here](https://github.com/ros2/rosbag2/blob/c611a8010a0aaaba8bff47c42846aae2374fe81f/rosbag2_cpp/include/rosbag2_cpp/cache/message_cache_buffer.hpp#L39), where it is interpreted as a limit on the size of the buffer in bytes.